config/configHelpers.ts

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+
+import merge from 'lodash.merge';
+import {
+  configurationType,
+  programmaticValue,
+  configurationLiteralValue,
+} from './types';
+
+/** helper function to get the value of an environment variable, or a defined fallback value */
+export const getEnvvarValue: (
+  name: string,
+  fallback?: configurationLiteralValue
+) => () => configurationLiteralValue = (
+  name,
+  fallback = `No value for ENVVAR ${name}`
+) => () => process.env[name] || fallback;
+
+/** helper used to process a `configurationDeclaration` or `featureFlagDeclaration` (D) to a `configurationType` with values being of type (T) for use across the codebase */
+export const processConfig: <D, T>(config: Array<D>) => configurationType<T> = (
+  config
+) => {
+  // merge all provided configs
+  const mergedConfig = config.reduce(
+    (acc, thisConfig) => merge(acc, thisConfig),
```

Comment:
Are there any of the following use cases expected:
1. Two configs have the same key and primitive value
2. Two configs have the same key with an array value


If so - merge will override value with the latest (in array case see : https://github.com/lodash/lodash/issues/2872)